packages/react/package.json

```diff
@@ -4,8 +4,14 @@
   "main": "dist/index.js",
   "module": "dist/esm/index.js",
   "exports": {
-    ".": "./dist/esm/index.js",
-    "./legacy": "./dist/esm/legacy.js",
+    ".": {
+      "import": "./dist/esm/index.js",
+      "require": "./dist/index.js"
+    },
+    "./legacy": {
```

Comment:
Does this require any changes to
https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/legacy/package.json?

It sounds like there's inconsistent behavior between environments:

- ESM bundlers (e.g. `esbuild`) respect `exports`
- CJS bundlers (some webpack configs) ignore `exports` and read `main`.
- Node, however, originally read `main` but newer versions read `exports` _even if not within a strict module structure_.

What I'm wondering is _how can we test or validate this_?